### ISSUE TITLE
encoding/csv: Writer preserves \r

### ISSUE THREAD

**Issue by zacharysyoung** [golang/go#58718]
### What version of Go are you using (`go version`)?

1.20

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

Playground

### What did you do?

Used the csv.Writer to write-out a string with a CRLF.

### What did you expect to see?

The same string, but with the CR dropped; just LF.

### What did you see instead?

The exact same string, the CRLF.

The [documentation](https://pkg.go.dev/encoding/csv#pkg-overview) makes the blanket assertion:

> Carriage returns before newline characters are silently removed.

Yet the single-field record, `[]string{" foo1 \r\n foo2 "}` will be written-out as `" foo1 \r\n foo2 "`, and not `" foo1 \n foo2 "` as I interpret the documentation (and #22746).

```go
package main

import (
	"encoding/csv"
	"fmt"
	"strings"
)

func main() {
	var b strings.Builder
	w := csv.NewWriter(&b)
	w.Write([]string{" foo1 \r\n foo2 "})
	w.Flush()

	want := " foo1 \n foo2 "
	got := b.String()
	got = got[1 : len(got)-2] // trim field quotes

	fmt.Printf("%q != %q", got, want)
}
```

Prints:

```none
" foo1 \r\n foo2 " != " foo1 \n foo2 "
```

I use the term "blanket assertion" because that documentation doesn't differentiate between reading and writing, so I expect—and especially after reading #22746—the csv package to effectively disallow a carriage return (CR, \r).

Does the documentation need clarification? Is this a bug with the Writer?

**Comment by seankhliao** [comment]
The entire section only applies to the input format (for Reader not Writer). There are clearly parts that can only be for reading (blank lines) and the examples also support that.

**Comment by zacharysyoung** [comment]
@seankhliao, I think you're saying the text I called out only applies to the Reader type; I'm not sure what you mean about blank lines...

- The text I called is only from third paragraph of the **Overview**, and both the words "read" and "write" precede that text exactly once.
- I see no language in preceding two paragraphs that limit the scope of meaning.
- @rsc's arguments in #22764 seem very principled, calling into question the place/value of carriage returns in a CSV, and putting that up against how he sees Go behaving.

I think there's still an issue.
